src/Ast.ml

```diff
@@ -50,6 +50,13 @@ let is_kwdopchar = function
       true
   | _ -> false
 
+let is_monadic_last_char = function '=' | '|' -> true | _ -> false
```

Comment:
I'm not familiar with the existing code but this makes me uneasy; what about `|>`?

Reply:
at first I wanted to avoid doing it for `|>` because it makes a long diff but it is more consistent indeed, fixed with my last commit.